File: strykerjs-node-test-poc/src/collections.ts
export function sum(xs: number[]): number {
  return xs.reduce((acc, x) => acc + x, 0);
}

export function average(xs: number[]): number | undefined {
  if (xs.length === 0) return undefined;
  return sum(xs) / xs.length;
}

export function evens(xs: number[]): number[] {
  return xs.filter((x) => x % 2 === 0);
}

export function hasNegative(xs: number[]): boolean {
  return xs.some((x) => x < 0);
}

export function allPositive(xs: number[]): boolean {
  return xs.every((x) => x > 0);
}

export function max(xs: number[]): number | undefined {
  if (xs.length === 0) return undefined;
  return Math.max(...xs);
}

export function dedupe(xs: number[]): number[] {
  return [...new Set(xs)];
}

export function take(xs: number[], n: number): number[] {
  return xs.slice(0, n);
}

File: stryker4s-poc/src/main/scala/poc/Collections.scala
package poc

object Collections {

  def sum(xs: List[Int]): Int = xs.sum

  def average(xs: List[Int]): Option[Double] =
    if (xs.isEmpty) None else Some(xs.sum.toDouble / xs.size)

  def evens(xs: List[Int]): List[Int] = xs.filter(_ % 2 == 0)

  def hasNegative(xs: List[Int]): Boolean = xs.exists(_ < 0)

  def allPositive(xs: List[Int]): Boolean = xs.forall(_ > 0)

  def max(xs: List[Int]): Option[Int] = xs.headOption.map(_ => xs.max)

  def dedupe(xs: List[Int]): List[Int] = xs.distinct

  def take(xs: List[Int], n: Int): List[Int] = xs.take(n)
}
